rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,44 @@
+---
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from Freenode to our own Matrix homeserver as our official
+community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network affects may be negative instead of
+positive.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to the new
+Matrix homeserver that has already been created which you can join by
+visting the [Invite
+link](https://app.element.io/#/room/#community:nixos.org).
```

Comment:
It's not really joining the server per-se. Maybe mention that you have to make an account elsewhere and then you can join the #community:nixos.org space? (Maybe this is getting into the semantics too much? Not sure how detailed these RFCs are supposed to be.)

Reply:
Having to find a suitable homeserver from which to participate is an obstacle that should be mentioned.

Reply:
Agree. That is a downside for sure. Maybe worth mentioning that matrix.org is at least decent these days?

Reply:
When you visit the invite link, it has a workflow where you can make an account on `matrix.org`. If you want to write some more detailed guide and put it on the NixOS wiki or something, I'd link to it!

Reply:
I've removed mentions of homeservers from the RFC, but if someone wants to come up with a self contained paragraph that talks about them, I'd like to include it.

Reply:
> Maybe worth mentioning that matrix.org is at least decent these days?

I'd think the exact opposite. Don't throw your (decentralized) eggs in one (centralized) basket.

Reply:
> I'd think the exact opposite. Don't throw your (decentralized) eggs in one (centralized) basket.

Well, I  would rather people get on matrix.org than not get on at all :shrug:

Reply:
I think it's worth mentioning that everyone being on `matrix.org` is about equivalent to the freenode status quo ante. A single person off `matrix.org` is more decentralized than freenode #nixos was.

Reply:
I think also the most important point is that each person can make that choice. If you are happy with `matrix.org` (as I am for now) you can use it. If you want to self host then you can and will be isolated from anything that happens on `matrix.org`. With the other proposed options that choice is made for you, whether that is Slack or Discord or a NixOS-hosted XMPP chat room.